### PR TITLE
Base64 decoder rust/v9

### PR DESCRIPTION
**Pull request by inashivb**
Link to tickets:
- https://redmine.openinfosecfoundation.org/issues/7065
- https://redmine.openinfosecfoundation.org/issues/7058
- https://redmine.openinfosecfoundation.org/issues/6280

Previous PR: https://github.com/OISF/suricata/pull/11542

Changes since v8:
- remove unnecessary copying to stack alloc'd array
- use heap alloc'd array as opposed to stack alloc'd array because we don't know the size of the decoded string so the stack alloc'd arrays were always much bigger than the actual decoded string but with heap allocation, we end up truncating the vector to exactly the decoded bytes and pass that back to C

### CODE REVIEW

File: rust/src/ffi/base64.rs [hunk 1]
```diff
@@ -26,6 +27,132 @@ pub enum Base64ReturnCode {
...
+
+/// Base64 decode a buffer.
+///
+/// This method exposes the Rust base64 decoder to C and should not be called from
+/// Rust code.
+///
+/// It allows decoding in the modes described by ``Base64Mode`` enum.
+#[no_mangle]
+pub unsafe extern "C" fn Base64Decode(
+    input: *const u8, len: u32, max_decoded: u32, mode: Base64Mode,
```
Comment (victorjulien) [inline rust/src/ffi/base64.rs:111]: what I expected here was a way to pass an existing buffer in, that would be filled by this function

Now we still allocate things

Comment (inashivb) [reply rust/src/ffi/base64.rs:111]: There is no longer a static buffer on the C side anymore. And, the vector allocated now has only the required size of the decoded string and not the length of the encoded string (which was the length of the static array).
1. Could you please tell why is it still better to pass a static buffer here to be filled in?
2. Do you have an example for that?

Thank you!

Comment (victorjulien) [reply rust/src/ffi/base64.rs:111]: The stack allocation doesn't exist anymore as this patch removes it :) I argue it should stay and be used instead.

Comment (victorjulien) [reply rust/src/ffi/base64.rs:111]: `rs_to_hex` is an example

Comment (inashivb) [reply rust/src/ffi/base64.rs:111]: Thank you. Attempting today.

**Pull request closed without merging**